srv/methods.go

```diff
@@ -59,38 +58,53 @@ type ResultGetTransaction struct {
 
 func (s *APIServer) getTransaction(getEntry bool) jrpc.MethodFunc {
 	return func(data json.RawMessage) interface{} {
-		//params := ParamsGetTransaction{}
-		//chain, put, err := validate(data, &params)
-		//if err != nil {
-		//	return err
-		//}
-		//defer put()
+		params := ParamsGetTransaction{}
+		_, _, err := validate(data, &params)
+		if err != nil {
+			return err
+		}
 
-		// TODO: use pegnet specific database code here to fetch an entry
-		//entry, err := entries.SelectValidByHash(chain.Conn, params.Hash)
-		//if err != nil {
-		//	panic(err)
-		//}
-		//if !entry.IsPopulated() {
-		//	return ErrorTransactionNotFound
-		//}
-		//
-		//if getEntry {
-		//	return entry
-		//}
-		//
-		//result := ResultGetTransaction{
-		//	Hash:      entry.Hash,
-		//	Timestamp: entry.Timestamp.Unix(),
-		//}
-		//
-		//tx := fat2.NewTransactionBatch(entry)
-		//if err := tx.UnmarshalEntry(); err != nil {
-		//	panic(err)
-		//}
-		//result.Tx = tx
-		//return result
-		return nil
+		found, err := s.Node.Pegnet.DoesTransactionExist(*params.Hash)
+		if err != nil {
+			return err
+		}
+
+		if !found {
+			return ErrorTransactionNotFound
+		}
+
+		var e factom.Entry
+		e.Hash = params.Hash
+		err = e.Get(s.Node.FactomClient)
+		if err != nil {
+			return jsonrpc2.InternalError
+		}
+
+		if !e.IsPopulated() {
+			return ErrorTransactionNotFound
+		}
+
+		var res ResultGetTransaction
+		res.Hash = e.Hash
+		// TODO: Save timestamp? We'd have to save it to the db
+		//res.Timestamp = e.Timestamp.Unix()
+		// TODO: Fill out the txid field
+		//res.TxIndex
+
+		if getEntry {
+			// TODO: What format do you want?
+			res.Tx = e
+			return e
```

Comment:
Yeah its just returning the entry, like this: https://github.com/Factom-Asset-Tokens/fatd/blob/develop/srv/methods.go#L123